be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -561,6 +561,29 @@ Status SegmentIterator::next_batch(RowBlockV2* block) {
             i += range_size;
         }
     }
+
+    // phase 4: read delete index, fill in the row whether is delete.
+    {
+        std::shared_ptr<Roaring> current_bitmap(new Roaring());
+    
+        // fetch delete index
+        const Roaring& delete_bitmap = _segment->delete_index_iterator().delete_bitmap();
+
+        const uint16_t* sv = block->selection_vector();
+        const uint16_t sv_size = block->selected_size();
+        uint16_t i = 0;
+        // check the delete rows and fill in the current_bitmap
+        // which use i in selection_vector as rowid 
+        while (i < sv_size) {
+
+            if(delete_bitmap.contains(_block_rowids[sv[i]])) {
+                current_bitmap->add(i);
+            }
+            i++;
```

Comment:
why not put this code in set_delete_bitmap?
it is not a good idea to pass shared_ptr